### ISSUE TITLE
Test get crushed because of error “ dangerouslyRenderMarkup ”

### ISSUE THREAD

**Issue by optimatex** [airbnb/enzyme#521]
The full error text is Invariant Violation: dangerouslyRenderMarkup(...): Cannot render markup in a worker thread. Make sure window and document are available globally before requiring React when unit testing or use ReactDOMServer.renderToString for server rendering.

I have searched this issue but all solutions are connected with jsdom set up. My jsdom set up is here :

```

import jsdom from 'jsdom';
import jquery from 'jquery';
import TestUtils from 'react-addons-test-utils';
import ReactDOM from 'react-dom';
import chai, { expect } from 'chai';
import React from 'react';
import chaiJquery from 'chai-jquery';

// Set up testing environment to run like a browser in the command line
global.document = jsdom.jsdom('<!doctype html><html><body></body></html>');
global.window = global.document.defaultView;
global.navigator = global.window.navigator;
const $ = jquery(global.window);

// build 'renderComponent' helper that should render a given react class
function renderComponent(ComponentClass, props, state) {
   const componentInstance = TestUtils.renderIntoDocument(
      <ComponentClass {...props} />
   );

   return $(ReactDOM.findDOMNode(componentInstance)); // produces HTML
}

// Build helper for simulating events
$.fn.simulate = function(eventName, value) {
   if (value) {
      this.val(value);
   }
   TestUtils.Simulate[eventName](this[0]);
}

// Set up chai-jquery
chaiJquery(chai, chai.util, $);

export { renderComponent, expect };

```

p.s. i don't use renderComponent function. It's lefted from previous approach

here is example of my test code

```

'use strict';
import React from 'react';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import chai, { expect } from 'chai';
import Profile_Store from '../profile.store';
import Signup from '../components/signup.component';

describe('__Signup__', ()=> {
   let signup_stub, spySubmit, spyCloseHint, wrap, context, email, pssword, passwordConfirm, submit, form_hint, input__hint;

   beforeEach(() => {
      spySubmit = sinon.spy(Signup.prototype, 'handleSubmit');
      signup_stub = sinon.stub(Profile_Store.prototype, 'signup');
      spyCloseHint = sinon.spy(Profile_Store.prototype, 'clearErrorMessage');
      context = {
         profile: new Profile_Store()
      };
      wrap = mount(<Signup  />, {context});
      email = wrap.find('input[type="email"]');
      pssword = wrap.find('input[type="password"]');
      submit = wrap.find('button[type="submit"]');
      input__hint = wrap.find('.input__hint');
      form_hint = wrap.find('.form__hint');
   });
   afterEach(() => {
      spySubmit.restore();
      spyCloseHint.restore();
      signup_stub.restore();
   });

   it("contains all inputs, button and validation hints", function() {
      expect(email).to.have.length(1);
      expect(pssword).to.have.length(2);
      expect(submit).to.have.length(1);
      expect(form_hint).to.have.length(1);
      expect(input__hint).to.have.length(3);
   });

   it("validation messages are hidden by default", function() {
      expect(form_hint.hasClass('hidden')).to.equal(true);
      input__hint.forEach((hint) => {
         expect(hint.hasClass('hidden')).to.equal(true);
      });
   });

   it("validation messages are visible if error", function() {
      context = {profile: new Profile_Store()};
      wrap = mount(<Signup  />, {context});
      wrap.context().profile.errorMessage = 'registration failed';
      expect(wrap.find('.form__hint').hasClass('visible')).to.equal(true);
      expect(wrap.find('.form__hint > span').at(0).text()).to.equal('registration failed');
   });

   it("validation messages are hidden while click close hint", function() {
      context.profile.errorMessage = 'registration failed';
      wrap = shallow(<Signup  />, {context});
      wrap.find('.form__closeHint').simulate('click');
      expect(context.profile.errorMessage).to.equal('');
      wrap = shallow(<Signup  />, {context});
      expect(wrap.find('.form__hint').hasClass('visible')).to.equal(false);
      sinon.assert.calledOnce(spyCloseHint);
   });

   it('form work correctly', () => {
      const event = {target: {value: 'cats'}};
      wrap.ref('email').simulate('change', event);
      wrap.simulate('submit',  { preventDefault() {} });
      chai.assert.equal(spySubmit.callCount, 1);
      sinon.assert.calledOnce(signup_stub);
   });

   it('don\'t handle Submit with empty values', () => {
      wrap.setState({
         validEmail: null,
         validPassword: null,
         validPasswordConfirm: false
      });
      wrap.simulate('submit',  { preventDefault() {} });
      sinon.assert.notCalled(signup_stub);
   });
});

```

**Comment by blainekasten** [comment]
React expects a DOM of some sort to be available to call `render` into for obvious reasons. Clearly something isn't quite right with your jsdom setup. Is this the first time writing tests with this setup or did this regression just appear? Can you do something like `console.log(window)` at the top (outside of tests) and see if it as actually a value?

**Comment by demoritik** [comment]
Hi. Sorry for long delay. I found that if i disable react waypoint https://github.com/brigade/react-waypoint - tests run errorless. But if it is in use - it cause that error . About your advice - i have include  console.log(windows) and here is output
```

Window {
  '$':
   { [Function]
     fn:
      { jquery: '2.2.4',
        constructor: [Circular],
        selector: '',
        length: 0,
        toArray: [Function],
        get: [Function],
        pushStack: [Function],
        each: [Function],
        map: [Function],
        slice: [Function],
        first: [Function],
        last: [Function],
        eq: [Function],
        end: [Function],
        push: [Function: push],
        sort: [Function: sort],
        splice: [Function: splice],
        extend: [Function],
        find: [Function],
        filter: [Function],
        not: [Function],
        is: [Function],
        init: [Function],
        has: [Function],
        closest: [Function],
        index: [Function],
        add: [Function],
        addBack: [Function],
        parent: [Function],
        parents: [Function],
        parentsUntil: [Function],
        next: [Function],
        prev: [Function],
        nextAll: [Function],
        prevAll: [Function],
        nextUntil: [Function],
        prevUntil: [Function],
        siblings: [Function],
        children: [Function],
        contents: [Function],
        ready: [Function],
        data: [Function],
        removeData: [Function],
        queue: [Function],
        dequeue: [Function],
        clearQueue: [Function],
        promise: [Function],
        on: [Function],
        one: [Function],
        off: [Function],
        domManip: [Function: domManip],
        detach: [Function],
        remove: [Function],
        text: [Function],
        append: [Function],
        prepend: [Function],
        before: [Function],
        after: [Function],
        empty: [Function],
        clone: [Function],
        html: [Function],
        replaceWith: [Function],
        appendTo: [Function],
        prependTo: [Function],
        insertBefore: [Function],
        insertAfter: [Function],
        replaceAll: [Function],
        css: [Function],
        show: [Function],
        hide: [Function],
        toggle: [Function],
        fadeTo: [Function],
        animate: [Function],
        stop: [Function],
        finish: [Function],
        slideDown: [Function],
        slideUp: [Function],
        slideToggle: [Function],
        fadeIn: [Function],
        fadeOut: [Function],
        fadeToggle: [Function],
        delay: [Function],
        attr: [Function],
        removeAttr: [Function],
        prop: [Function],
        removeProp: [Function],
        addClass: [Function],
        removeClass: [Function],
        toggleClass: [Function],
        hasClass: [Function],
        val: [Function],
        trigger: [Function],
        triggerHandler: [Function],
        blur: [Function],
        focus: [Function],
        focusin: [Function],
        focusout: [Function],
        load: [Function],
        resize: [Function],
        scroll: [Function],
        unload: [Function],
        click: [Function],
        dblclick: [Function],
        mousedown: [Function],
        mouseup: [Function],
        mousemove: [Function],
        mouseover: [Function],
        mouseout: [Function],
        mouseenter: [Function],
        mouseleave: [Function],
        change: [Function],
        select: [Function],
        submit: [Function],
        keydown: [Function],
        keypress: [Function],
        keyup: [Function],
        error: [Function],
        contextmenu: [Function],
        hover: [Function],
        wrapAll: [Function],
        wrapInner: [Function],
        wrap: [Function],
        unwrap: [Function],
        serialize: [Function],
        serializeArray: [Function],
        ajaxStart: [Function],
        ajaxStop: [Function],
        ajaxComplete: [Function],
        ajaxError: [Function],
        ajaxSuccess: [Function],
        ajaxSend: [Function],
        offset: [Function],
        position: [Function],
        offsetParent: [Function],
        scrollLeft: [Function],
        scrollTop: [Function],
        innerHeight: [Function],
        height: [Function],
        outerHeight: [Function],
        innerWidth: [Function],
        width: [Function],
        outerWidth: [Function],
        bind: [Function],
        unbind: [Function],
        delegate: [Function],
        undelegate: [Function],
        size: [Function],
        andSelf: [Function],
        inspect: [Function],
        simulate: [Function] },
     extend: [Function],
     expando: 'jQuery22406217277268879116',
     isReady: true,
     error: [Function],
     noop: [Function],
     isFunction: [Function],
     isArray: [Function: isArray],
     isWindow: [Function],
     isNumeric: [Function],
     isPlainObject: [Function],
     isEmptyObject: [Function],
     type: [Function],
     globalEval: [Function],
     camelCase: [Function],
     nodeName: [Function],
     each: [Function],
     trim: [Function],
     makeArray: [Function],
     inArray: [Function],
     merge: [Function],
     grep: [Function],
     map: [Function],
     guid: 1,
     proxy: [Function],
     now: [Function: now],
     support:
      { checkClone: true,
        noCloneChecked: true,
        clearCloneStyle: true,
        pixelPosition: [Function],
        boxSizingReliable: [Function],
        pixelMarginRight: [Function],
        reliableMarginLeft: [Function],
        reliableMarginRight: [Function],
        checkOn: false,
        optSelected: true,
        optDisabled: true,
        radioValue: true,
        focusin: false,
        cors: true,
        ajax: true },
     find:
      { [Function: Sizzle]
        support: [Object],
        isXML: [Function],
        setDocument: [Function],
        matches: [Function],
        matchesSelector: [Function],
        contains: [Function],
        attr: [Function],
        error: [Function],
        uniqueSort: [Function],
        getText: [Function],
        selectors: [Object],
        tokenize: [Function],
        compile: [Function],
        select: [Function] },
     expr:
      { cacheLength: 50,
        createPseudo: [Function: markFunction],
        match: [Object],
        attrHandle: [Object],
        find: [Object],
        relative: [Object],
        preFilter: [Object],
        filter: [Object],
        pseudos: [Object],
        filters: [Object],
        setFilters: {},
        ':': [Object] },
     unique: [Function],
     uniqueSort: [Function],
     text: [Function],
     isXMLDoc: [Function],
     contains: [Function],
     filter: [Function],
     Callbacks: [Function],
     Deferred: [Function],
     when: [Function],
     readyWait: 0,
     holdReady: [Function],
     ready: { [Function] promise: [Function] },
     hasData: [Function],
     data: [Function],
     removeData: [Function],
     _data: [Function],
     _removeData: [Function],
     queue: [Function],
     dequeue: [Function],
     _queueHooks: [Function],
     event:
      { global: {},
        add: [Function],
        remove: [Function],
        dispatch: [Function],
        handlers: [Function],
        props: [Object],
        fixHooks: {},
        keyHooks: [Object],
        mouseHooks: [Object],
        fix: [Function],
        special: [Object],
        trigger: [Function],
        simulate: [Function] },
     removeEvent: [Function],
     Event: [Function],
     htmlPrefilter: [Function],
     clone: [Function],
     cleanData: [Function],
     cssHooks:
      { opacity: [Object],
        height: [Object],
        width: [Object],
        marginLeft: [Object],
        marginRight: [Object],
        margin: [Object],
        padding: [Object],
        borderWidth: [Object],
        top: [Object],
        left: [Object] },
     cssNumber:
      { animationIterationCount: true,
        columnCount: true,
        fillOpacity: true,
        flexGrow: true,
        flexShrink: true,
        fontWeight: true,
        lineHeight: true,
        opacity: true,
        order: true,
        orphans: true,
        widows: true,
        zIndex: true,
        zoom: true },
     cssProps: { float: 'cssFloat' },
     style: [Function],
     css: [Function],
     Tween: { [Function: Tween] propHooks: [Object] },
     easing: { linear: [Function], swing: [Function], _default: 'swing' },
     fx:
      { [Function]
        step: {},
        tick: [Function],
        timer: [Function],
        interval: 13,
        start: [Function],
        stop: [Function],
        speeds: [Object] },
     Animation:
      { [Function: Animation]
        tweeners: [Object],
        tweener: [Function],
        prefilters: [Object],
        prefilter: [Function] },
     speed: [Function],
     timers: [],
     attr: [Function],
     attrHooks: { type: [Object] },
     removeAttr: [Function],
     prop: [Function],
     propHooks: { tabIndex: [Object] },
     propFix:
      { for: 'htmlFor',
        class: 'className',
        tabindex: [String: 'tabIndex'],
        readonly: [String: 'readOnly'],
        maxlength: [String: 'maxLength'],
        cellspacing: [String: 'cellSpacing'],
        cellpadding: [String: 'cellPadding'],
        rowspan: [String: 'rowSpan'],
        colspan: [String: 'colSpan'],
        usemap: [String: 'useMap'],
        frameborder: [String: 'frameBorder'],
        contenteditable: [String: 'contentEditable'] },
     valHooks:
      { option: [Object],
        select: [Object],
        radio: [Object],
        checkbox: [Object] },
     parseJSON: [Function],
     parseXML: [Function],
     active: 0,
     lastModified: {},
     etag: {},
     ajaxSettings:
      { url: 'about:blank',
        type: 'GET',
        isLocal: true,
        global: true,
        processData: true,
        async: true,
        contentType: 'application/x-www-form-urlencoded; charset=UTF-8',
        accepts: [Object],
        contents: [Object],
        responseFields: [Object],
        converters: [Object],
        flatOptions: [Object],
        xhr: [Function],
        jsonp: 'callback',
        jsonpCallback: [Function] },
     ajaxSetup: [Function],
     ajaxPrefilter: [Function],
     ajaxTransport: [Function],
     ajax: [Function],
     getJSON: [Function],
     getScript: [Function],
     get: [Function],
     post: [Function],
     _evalUrl: [Function],
     param: [Function],
     parseHTML: [Function],
     offset: { setOffset: [Function] },
     noConflict: [Function],
     ua:
      { ua: 'Node.js (win32; U; rv:v4.4.5)',
        browser: [Object],
        engine: [Object],
        os: [Object],
        device: [Object],
        cpu: [Object],
        get: [Function],
        set: [Function] } },
  jQuery:
   { [Function]
     fn:
      { jquery: '2.2.4',
        constructor: [Circular],
        selector: '',
        length: 0,
        toArray: [Function],
        get: [Function],
        pushStack: [Function],
        each: [Function],
        map: [Function],
        slice: [Function],
        first: [Function],
        last: [Function],
        eq: [Function],
        end: [Function],
        push: [Function: push],
        sort: [Function: sort],
        splice: [Function: splice],
        extend: [Function],
        find: [Function],
        filter: [Function],
        not: [Function],
        is: [Function],
        init: [Function],
        has: [Function],
        closest: [Function],
        index: [Function],
        add: [Function],
        addBack: [Function],
        parent: [Function],
        parents: [Function],
        parentsUntil: [Function],
        next: [Function],
        prev: [Function],
        nextAll: [Function],
        prevAll: [Function],
        nextUntil: [Function],
        prevUntil: [Function],
        siblings: [Function],
        children: [Function],
        contents: [Function],
        ready: [Function],
        data: [Function],
        removeData: [Function],
        queue: [Function],
        dequeue: [Function],
        clearQueue: [Function],
        promise: [Function],
        on: [Function],
        one: [Function],
        off: [Function],
        domManip: [Function: domManip],
        detach: [Function],
        remove: [Function],
        text: [Function],
        append: [Function],
        prepend: [Function],
        before: [Function],
        after: [Function],
        empty: [Function],
        clone: [Function],
        html: [Function],
        replaceWith: [Function],
        appendTo: [Function],
        prependTo: [Function],
        insertBefore: [Function],
        insertAfter: [Function],
        replaceAll: [Function],
        css: [Function],
        show: [Function],
        hide: [Function],
        toggle: [Function],
        fadeTo: [Function],
        animate: [Function],
        stop: [Function],
        finish: [Function],
        slideDown: [Function],
        slideUp: [Function],
        slideToggle: [Function],
        fadeIn: [Function],
        fadeOut: [Function],
        fadeToggle: [Function],
        delay: [Function],
        attr: [Function],
        removeAttr: [Function],
        prop: [Function],
        removeProp: [Function],
        addClass: [Function],
        removeClass: [Function],
        toggleClass: [Function],
        hasClass: [Function],
        val: [Function],
        trigger: [Function],
        triggerHandler: [Function],
        blur: [Function],
        focus: [Function],
        focusin: [Function],
        focusout: [Function],
        load: [Function],
        resize: [Function],
        scroll: [Function],
        unload: [Function],
        click: [Function],
        dblclick: [Function],
        mousedown: [Function],
        mouseup: [Function],
        mousemove: [Function],
        mouseover: [Function],
        mouseout: [Function],
        mouseenter: [Function],
        mouseleave: [Function],
        change: [Function],
        select: [Function],
        submit: [Function],
        keydown: [Function],
        keypress: [Function],
        keyup: [Function],
        error: [Function],
        contextmenu: [Function],
        hover: [Function],
        wrapAll: [Function],
        wrapInner: [Function],
        wrap: [Function],
        unwrap: [Function],
        serialize: [Function],
        serializeArray: [Function],
        ajaxStart: [Function],
        ajaxStop: [Function],
        ajaxComplete: [Function],
        ajaxError: [Function],
        ajaxSuccess: [Function],
        ajaxSend: [Function],
        offset: [Function],
        position: [Function],
        offsetParent: [Function],
        scrollLeft: [Function],
        scrollTop: [Function],
        innerHeight: [Function],
        height: [Function],
        outerHeight: [Function],
        innerWidth: [Function],
        width: [Function],
        outerWidth: [Function],
        bind: [Function],
        unbind: [Function],
        delegate: [Function],
        undelegate: [Function],
        size: [Function],
        andSelf: [Function],
        inspect: [Function],
        simulate: [Function] },
     extend: [Function],
     expando: 'jQuery22406217277268879116',
     isReady: true,
     error: [Function],
     noop: [Function],
     isFunction: [Function],
     isArray: [Function: isArray],
     isWindow: [Function],
     isNumeric: [Function],
     isPlainObject: [Function],
     isEmptyObject: [Function],
     type: [Function],
     globalEval: [Function],
     camelCase: [Function],
     nodeName: [Function],
     each: [Function],
     trim: [Function],
     makeArray: [Function],
     inArray: [Function],
     merge: [Function],
     grep: [Function],
     map: [Function],
     guid: 1,
     proxy: [Function],
     now: [Function: now],
     support:
      { checkClone: true,
        noCloneChecked: true,
        clearCloneStyle: true,
        pixelPosition: [Function],
        boxSizingReliable: [Function],
        pixelMarginRight: [Function],
        reliableMarginLeft: [Function],
        reliableMarginRight: [Function],
        checkOn: false,
        optSelected: true,
        optDisabled: true,
        radioValue: true,
        focusin: false,
        cors: true,
        ajax: true },
     find:
      { [Function: Sizzle]
        support: [Object],
        isXML: [Function],
        setDocument: [Function],
        matches: [Function],
        matchesSelector: [Function],
        contains: [Function],
        attr: [Function],
        error: [Function],
        uniqueSort: [Function],
        getText: [Function],
        selectors: [Object],
        tokenize: [Function],
        compile: [Function],
        select: [Function] },
     expr:
      { cacheLength: 50,
        createPseudo: [Function: markFunction],
        match: [Object],
        attrHandle: [Object],
        find: [Object],
        relative: [Object],
        preFilter: [Object],
        filter: [Object],
        pseudos: [Object],
        filters: [Object],
        setFilters: {},
        ':': [Object] },
     unique: [Function],
     uniqueSort: [Function],
     text: [Function],
     isXMLDoc: [Function],
     contains: [Function],
     filter: [Function],
     Callbacks: [Function],
     Deferred: [Function],
     when: [Function],
     readyWait: 0,
     holdReady: [Function],
     ready: { [Function] promise: [Function] },
     hasData: [Function],
     data: [Function],
     removeData: [Function],
     _data: [Function],
     _removeData: [Function],
     queue: [Function],
     dequeue: [Function],
     _queueHooks: [Function],
     event:
      { global: {},
        add: [Function],
        remove: [Function],
        dispatch: [Function],
        handlers: [Function],
        props: [Object],
        fixHooks: {},
        keyHooks: [Object],
        mouseHooks: [Object],
        fix: [Function],
        special: [Object],
        trigger: [Function],
        simulate: [Function] },
     removeEvent: [Function],
     Event: [Function],
     htmlPrefilter: [Function],
     clone: [Function],
     cleanData: [Function],
     cssHooks:
      { opacity: [Object],
        height: [Object],
        width: [Object],
        marginLeft: [Object],
        marginRight: [Object],
        margin: [Object],
        padding: [Object],
        borderWidth: [Object],
        top: [Object],
        left: [Object] },
     cssNumber:
      { animationIterationCount: true,
        columnCount: true,
        fillOpacity: true,
        flexGrow: true,
        flexShrink: true,
        fontWeight: true,
        lineHeight: true,
        opacity: true,
        order: true,
        orphans: true,
        widows: true,
        zIndex: true,
        zoom: true },
     cssProps: { float: 'cssFloat' },
     style: [Function],
     css: [Function],
     Tween: { [Function: Tween] propHooks: [Object] },
     easing: { linear: [Function], swing: [Function], _default: 'swing' },
     fx:
      { [Function]
        step: {},
        tick: [Function],
        timer: [Function],
        interval: 13,
        start: [Function],
        stop: [Function],
        speeds: [Object] },
     Animation:
      { [Function: Animation]
        tweeners: [Object],
        tweener: [Function],
        prefilters: [Object],
        prefilter: [Function] },
     speed: [Function],
     timers: [],
     attr: [Function],
     attrHooks: { type: [Object] },
     removeAttr: [Function],
     prop: [Function],
     propHooks: { tabIndex: [Object] },
     propFix:
      { for: 'htmlFor',
        class: 'className',
        tabindex: [String: 'tabIndex'],
        readonly: [String: 'readOnly'],
        maxlength: [String: 'maxLength'],
        cellspacing: [String: 'cellSpacing'],
        cellpadding: [String: 'cellPadding'],
        rowspan: [String: 'rowSpan'],
        colspan: [String: 'colSpan'],
        usemap: [String: 'useMap'],
        frameborder: [String: 'frameBorder'],
        contenteditable: [String: 'contentEditable'] },
     valHooks:
      { option: [Object],
        select: [Object],
        radio: [Object],
        checkbox: [Object] },
     parseJSON: [Function],
     parseXML: [Function],
     active: 0,
     lastModified: {},
     etag: {},
     ajaxSettings:
      { url: 'about:blank',
        type: 'GET',
        isLocal: true,
        global: true,
        processData: true,
        async: true,
        contentType: 'application/x-www-form-urlencoded; charset=UTF-8',
        accepts: [Object],
        contents: [Object],
        responseFields: [Object],
        converters: [Object],
        flatOptions: [Object],
        xhr: [Function],
        jsonp: 'callback',
        jsonpCallback: [Function] },
     ajaxSetup: [Function],
     ajaxPrefilter: [Function],
     ajaxTransport: [Function],
     ajax: [Function],
     getJSON: [Function],
     getScript: [Function],
     get: [Function],
     post: [Function],
     _evalUrl: [Function],
     param: [Function],
     parseHTML: [Function],
     offset: { setOffset: [Function] },
     noConflict: [Function],
     ua:
      { ua: 'Node.js (win32; U; rv:v4.4.5)',
        browser: [Object],
        engine: [Object],
        os: [Object],
        device: [Object],
        cpu: [Object],
        get: [Function],
        set: [Function] } },
  DOMException:
   { [Function: DOMException]
     INDEX_SIZE_ERR: 1,
     HIERARCHY_REQUEST_ERR: 3,
     WRONG_DOCUMENT_ERR: 4,
     INVALID_CHARACTER_ERR: 5,
     NO_MODIFICATION_ALLOWED_ERR: 7,
     NOT_FOUND_ERR: 8,
     NOT_SUPPORTED_ERR: 9,
     INUSE_ATTRIBUTE_ERR: 10,
     INVALID_STATE_ERR: 11,
     SYNTAX_ERR: 12,
     INVALID_MODIFICATION_ERR: 13,
     NAMESPACE_ERR: 14,
     INVALID_ACCESS_ERR: 15,
     SECURITY_ERR: 18,
     NETWORK_ERR: 19,
     ABORT_ERR: 20,
     URL_MISMATCH_ERR: 21,
     QUOTA_EXCEEDED_ERR: 22,
     TIMEOUT_ERR: 23,
     INVALID_NODE_TYPE_ERR: 24,
     DATA_CLONE_ERR: 25 },
  NamedNodeMap: [Function: NamedNodeMap],
  Attr: [Function: Attr],
  Node:
   { [Function: Node]
     ELEMENT_NODE: 1,
     ATTRIBUTE_NODE: 2,
     TEXT_NODE: 3,
     CDATA_SECTION_NODE: 4,
     ENTITY_REFERENCE_NODE: 5,
     ENTITY_NODE: 6,
     PROCESSING_INSTRUCTION_NODE: 7,
     COMMENT_NODE: 8,
     DOCUMENT_NODE: 9,
     DOCUMENT_TYPE_NODE: 10,
     DOCUMENT_FRAGMENT_NODE: 11,
     NOTATION_NODE: 12,
     DOCUMENT_POSITION_DISCONNECTED: 1,
     DOCUMENT_POSITION_PRECEDING: 2,
     DOCUMENT_POSITION_FOLLOWING: 4,
     DOCUMENT_POSITION_CONTAINS: 8,
     DOCUMENT_POSITION_CONTAINED_BY: 16,
     DOCUMENT_POSITION_IMPLEMENTATION_SPECIFIC: 32 },
  Element: [Function: Element],
  DocumentFragment: [Function: DocumentFragment],
  HTMLDocument: [Function: Document],
  Document: [Function: Document],
  CharacterData: [Function: CharacterData],
  Comment: [Function: Comment],
  DocumentType: [Function: DocumentType],
  DOMImplementation: [Function: DOMImplementation],
  ProcessingInstruction: [Function: ProcessingInstruction],
  Text: [Function: Text],
  Event: { [Function: Event] NONE: 0, CAPTURING_PHASE: 1, AT_TARGET: 2, BUBBLING_PHASE: 3 },
  CustomEvent: [Function: CustomEvent],
  MessageEvent: [Function: MessageEvent],
  ErrorEvent: [Function: ErrorEvent],
  HashChangeEvent: [Function: HashChangeEvent],
  PopStateEvent: [Function: PopStateEvent],
  UIEvent: [Function: UIEvent],
  MouseEvent: [Function: MouseEvent],
  KeyboardEvent:
   { [Function: KeyboardEvent]
     DOM_KEY_LOCATION_STANDARD: 0,
     DOM_KEY_LOCATION_LEFT: 1,
     DOM_KEY_LOCATION_RIGHT: 2,
     DOM_KEY_LOCATION_NUMPAD: 3 },
  TouchEvent: [Function: TouchEvent],
  MutationEvent: { [Function: MutationEvent] MODIFICATION: 1, ADDITION: 2, REMOVAL: 3 },
  ProgressEvent: [Function: ProgressEvent],
  EventTarget: [Function: EventTarget],
  Location: [Function: Location],
  History: [Function: History],
  HTMLElement: [Function: HTMLElement],
  HTMLAnchorElement: [Function: HTMLAnchorElement],
  HTMLAppletElement: [Function: HTMLAppletElement],
  HTMLAreaElement: [Function: HTMLAreaElement],
  HTMLAudioElement: [Function: HTMLAudioElement],
  HTMLBaseElement: [Function: HTMLBaseElement],
  HTMLBodyElement: [Function: HTMLBodyElement],
  HTMLBRElement: [Function: HTMLBRElement],
  HTMLButtonElement: [Function: HTMLButtonElement],
  HTMLCanvasElement: [Function: HTMLCanvasElement],
  HTMLDataElement: [Function: HTMLDataElement],
  HTMLDataListElement: [Function: HTMLDataListElement],
  HTMLDialogElement: [Function: HTMLDialogElement],
  HTMLDirectoryElement: [Function: HTMLDirectoryElement],
  HTMLDivElement: [Function: HTMLDivElement],
  HTMLDListElement: [Function: HTMLDListElement],
  HTMLEmbedElement: [Function: HTMLEmbedElement],
  HTMLFieldSetElement: [Function: HTMLFieldSetElement],
  HTMLFontElement: [Function: HTMLFontElement],
  HTMLFormElement: [Function: HTMLFormElement],
  HTMLFrameElement: [Function: HTMLFrameElement],
  HTMLFrameSetElement: [Function: HTMLFrameSetElement],
  HTMLHeadingElement: [Function: HTMLHeadingElement],
  HTMLHeadElement: [Function: HTMLHeadElement],
  HTMLHRElement: [Function: HTMLHRElement],
  HTMLHtmlElement: [Function: HTMLHtmlElement],
  HTMLIFrameElement: [Function: HTMLIFrameElement],
  HTMLImageElement: [Function: HTMLImageElement],
  HTMLInputElement: [Function: HTMLInputElement],
  HTMLLabelElement: [Function: HTMLLabelElement],
  HTMLLegendElement: [Function: HTMLLegendElement],
  HTMLLIElement: [Function: HTMLLIElement],
  HTMLLinkElement: [Function: HTMLLinkElement],
  HTMLMapElement: [Function: HTMLMapElement],
  HTMLMediaElement:
   { [Function: HTMLMediaElement]
     NETWORK_EMPTY: 0,
     NETWORK_IDLE: 1,
     NETWORK_LOADING: 2,
     NETWORK_NO_SOURCE: 3,
     HAVE_NOTHING: 0,
     HAVE_METADATA: 1,
     HAVE_CURRENT_DATA: 2,
     HAVE_FUTURE_DATA: 3,
     HAVE_ENOUGH_DATA: 4 },
  HTMLMenuElement: [Function: HTMLMenuElement],
  HTMLMetaElement: [Function: HTMLMetaElement],
  HTMLMeterElement: [Function: HTMLMeterElement],
  HTMLModElement: [Function: HTMLModElement],
  HTMLObjectElement: [Function: HTMLObjectElement],
  HTMLOListElement: [Function: HTMLOListElement],
  HTMLOptGroupElement: [Function: HTMLOptGroupElement],
  HTMLOptionElement: [Function: HTMLOptionElement],
  HTMLOutputElement: [Function: HTMLOutputElement],
  HTMLParagraphElement: [Function: HTMLParagraphElement],
  HTMLParamElement: [Function: HTMLParamElement],
  HTMLPreElement: [Function: HTMLPreElement],
  HTMLProgressElement: [Function: HTMLProgressElement],
  HTMLQuoteElement: [Function: HTMLQuoteElement],
  HTMLScriptElement: [Function: HTMLScriptElement],
  HTMLSelectElement: [Function: HTMLSelectElement],
  HTMLSourceElement: [Function: HTMLSourceElement],
  HTMLSpanElement: [Function: HTMLSpanElement],
  HTMLStyleElement: [Function: HTMLStyleElement],
  HTMLTableCaptionElement: [Function: HTMLTableCaptionElement],
  HTMLTableCellElement: [Function: HTMLTableCellElement],
  HTMLTableColElement: [Function: HTMLTableColElement],
  HTMLTableDataCellElement: [Function: HTMLTableDataCellElement],
  HTMLTableElement: [Function: HTMLTableElement],
  HTMLTableHeaderCellElement: [Function: HTMLTableHeaderCellElement],
  HTMLTimeElement: [Function: HTMLTimeElement],
  HTMLTitleElement: [Function: HTMLTitleElement],
  HTMLTableRowElement: [Function: HTMLTableRowElement],
  HTMLTableSectionElement: [Function: HTMLTableSectionElement],
  HTMLTemplateElement: [Function: HTMLTemplateElement],
  HTMLTextAreaElement: [Function: HTMLTextAreaElement],
  HTMLTrackElement: { [Function: HTMLTrackElement] NONE: 0, LOADING: 1, LOADED: 2, ERROR: 3 },
  HTMLUListElement: [Function: HTMLUListElement],
  HTMLUnknownElement: [Function: HTMLUnknownElement],
  HTMLVideoElement: [Function: HTMLVideoElement],
  StyleSheet: [Function: StyleSheet],
  MediaList: [Function: MediaList],
  CSSStyleSheet: [Function: CSSStyleSheet],
  CSSRule:
   { [Function: CSSRule]
     UNKNOWN_RULE: 0,
     STYLE_RULE: 1,
     CHARSET_RULE: 2,
     IMPORT_RULE: 3,
     MEDIA_RULE: 4,
     FONT_FACE_RULE: 5,
     PAGE_RULE: 6,
     KEYFRAMES_RULE: 7,
     KEYFRAME_RULE: 8,
     MARGIN_RULE: 9,
     NAMESPACE_RULE: 10,
     COUNTER_STYLE_RULE: 11,
     SUPPORTS_RULE: 12,
     DOCUMENT_RULE: 13,
     FONT_FEATURE_VALUES_RULE: 14,
     VIEWPORT_RULE: 15,
     REGION_STYLE_RULE: 16 },
  CSSStyleRule: { [Function: CSSStyleRule] parse: [Function] },
  CSSMediaRule: [Function: CSSMediaRule],
  CSSImportRule: [Function: CSSImportRule],
  CSSStyleDeclaration: [Function: CSSStyleDeclaration],
  StyleSheetList: [Function: StyleSheetList],
  XPathException: { [Function: XPathException] INVALID_EXPRESSION_ERR: 51, TYPE_ERR: 52 },
  XPathExpression: [Function: XPathExpression],
  XPathResult:
   { [Function: XPathResult]
     ANY_TYPE: 0,
     NUMBER_TYPE: 1,
     STRING_TYPE: 2,
     BOOLEAN_TYPE: 3,
     UNORDERED_NODE_ITERATOR_TYPE: 4,
     ORDERED_NODE_ITERATOR_TYPE: 5,
     UNORDERED_NODE_SNAPSHOT_TYPE: 6,
     ORDERED_NODE_SNAPSHOT_TYPE: 7,
     ANY_UNORDERED_NODE_TYPE: 8,
     FIRST_ORDERED_NODE_TYPE: 9 },
  XPathEvaluator: [Function: XPathEvaluator],
  HTMLCollection: [Function: HTMLCollection],
  NodeFilter:
   { [Function]
     acceptNode: [Function],
     SHOW_ALL: 4294967295,
     SHOW_ELEMENT: 1,
     SHOW_ATTRIBUTE: 2,
     SHOW_TEXT: 4,
     SHOW_CDATA_SECTION: 8,
     SHOW_ENTITY_REFERENCE: 16,
     SHOW_ENTITY: 32,
     SHOW_PROCESSING_INSTRUCTION: 64,
     SHOW_COMMENT: 128,
     SHOW_DOCUMENT: 256,
     SHOW_DOCUMENT_TYPE: 512,
     SHOW_DOCUMENT_FRAGMENT: 1024,
     SHOW_NOTATION: 2048,
     FILTER_ACCEPT: 1,
     FILTER_REJECT: 2,
     FILTER_SKIP: 3 },
  NodeIterator: [Function: NodeIterator],
  NodeList: [Function: NodeList],
  Blob: [Function: Blob],
  File: [Function: File],
  FileList: [Function: FileList],
  FormData: [Function: FormData],
  XMLHttpRequestEventTarget: [Function: XMLHttpRequestEventTarget],
  XMLHttpRequestUpload: [Function: XMLHttpRequestUpload],
  DOMTokenList: [Function: DOMTokenList],
  URL:
   { [Function: URL]
     domainToASCII: [Function: domainToASCII],
     domainToUnicode: [Function: domainToUnicode] },
  Window: [Function: Window],
  _core:
   { DOMException:
      { [Function: DOMException]
        INDEX_SIZE_ERR: 1,
        HIERARCHY_REQUEST_ERR: 3,
        WRONG_DOCUMENT_ERR: 4,
        INVALID_CHARACTER_ERR: 5,
        NO_MODIFICATION_ALLOWED_ERR: 7,
        NOT_FOUND_ERR: 8,
        NOT_SUPPORTED_ERR: 9,
        INUSE_ATTRIBUTE_ERR: 10,
        INVALID_STATE_ERR: 11,
        SYNTAX_ERR: 12,
        INVALID_MODIFICATION_ERR: 13,
        NAMESPACE_ERR: 14,
        INVALID_ACCESS_ERR: 15,
        SECURITY_ERR: 18,
        NETWORK_ERR: 19,
        ABORT_ERR: 20,
        URL_MISMATCH_ERR: 21,
        QUOTA_EXCEEDED_ERR: 22,
        TIMEOUT_ERR: 23,
        INVALID_NODE_TYPE_ERR: 24,
        DATA_CLONE_ERR: 25 },
     NamedNodeMap: [Function: NamedNodeMap],
     Attr: [Function: Attr],
     Node:
      { [Function: Node]
        ELEMENT_NODE: 1,
        ATTRIBUTE_NODE: 2,
        TEXT_NODE: 3,
        CDATA_SECTION_NODE: 4,
        ENTITY_REFERENCE_NODE: 5,
        ENTITY_NODE: 6,
        PROCESSING_INSTRUCTION_NODE: 7,
        COMMENT_NODE: 8,
        DOCUMENT_NODE: 9,
        DOCUMENT_TYPE_NODE: 10,
        DOCUMENT_FRAGMENT_NODE: 11,
        NOTATION_NODE: 12,
        DOCUMENT_POSITION_DISCONNECTED: 1,
        DOCUMENT_POSITION_PRECEDING: 2,
        DOCUMENT_POSITION_FOLLOWING: 4,
        DOCUMENT_POSITION_CONTAINS: 8,
        DOCUMENT_POSITION_CONTAINED_BY: 16,
        DOCUMENT_POSITION_IMPLEMENTATION_SPECIFIC: 32 },
     Element: [Function: Element],
     DocumentFragment: [Function: DocumentFragment],
     HTMLDocument: [Function: Document],
     Document: [Function: Document],
     CharacterData: [Function: CharacterData],
     Comment: [Function: Comment],
     DocumentType: [Function: DocumentType],
     DOMImplementation: [Function: DOMImplementation],
     ProcessingInstruction: [Function: ProcessingInstruction],
     Text: [Function: Text],
     Event: { [Function: Event] NONE: 0, CAPTURING_PHASE: 1, AT_TARGET: 2, BUBBLING_PHASE: 3 },
     CustomEvent: [Function: CustomEvent],
     MessageEvent: [Function: MessageEvent],
     ErrorEvent: [Function: ErrorEvent],
     HashChangeEvent: [Function: HashChangeEvent],
     PopStateEvent: [Function: PopStateEvent],
     UIEvent: [Function: UIEvent],
     MouseEvent: [Function: MouseEvent],
     KeyboardEvent:
      { [Function: KeyboardEvent]
        DOM_KEY_LOCATION_STANDARD: 0,
        DOM_KEY_LOCATION_LEFT: 1,
        DOM_KEY_LOCATION_RIGHT: 2,
        DOM_KEY_LOCATION_NUMPAD: 3 },
     TouchEvent: [Function: TouchEvent],
     MutationEvent: { [Function: MutationEvent] MODIFICATION: 1, ADDITION: 2, REMOVAL: 3 },
     ProgressEvent: [Function: ProgressEvent],
     EventTarget: [Function: EventTarget],
     Location: [Function: Location],
     History: [Function: History],
     HTMLElement: [Function: HTMLElement],
     HTMLAnchorElement: [Function: HTMLAnchorElement],
     HTMLAppletElement: [Function: HTMLAppletElement],
     HTMLAreaElement: [Function: HTMLAreaElement],
     HTMLAudioElement: [Function: HTMLAudioElement],
     HTMLBaseElement: [Function: HTMLBaseElement],
     HTMLBodyElement: [Function: HTMLBodyElement],
     HTMLBRElement: [Function: HTMLBRElement],
     HTMLButtonElement: [Function: HTMLButtonElement],
     HTMLCanvasElement: [Function: HTMLCanvasElement],
     HTMLDataElement: [Function: HTMLDataElement],
     HTMLDataListElement: [Function: HTMLDataListElement],
     HTMLDialogElement: [Function: HTMLDialogElement],
     HTMLDirectoryElement: [Function: HTMLDirectoryElement],
     HTMLDivElement: [Function: HTMLDivElement],
     HTMLDListElement: [Function: HTMLDListElement],
     HTMLEmbedElement: [Function: HTMLEmbedElement],
     HTMLFieldSetElement: [Function: HTMLFieldSetElement],
     HTMLFontElement: [Function: HTMLFontElement],
     HTMLFormElement: [Function: HTMLFormElement],
     HTMLFrameElement: [Function: HTMLFrameElement],
     HTMLFrameSetElement: [Function: HTMLFrameSetElement],
     HTMLHeadingElement: [Function: HTMLHeadingElement],
     HTMLHeadElement: [Function: HTMLHeadElement],
     HTMLHRElement: [Function: HTMLHRElement],
     HTMLHtmlElement: [Function: HTMLHtmlElement],
     HTMLIFrameElement: [Function: HTMLIFrameElement],
     HTMLImageElement: [Function: HTMLImageElement],
     HTMLInputElement: [Function: HTMLInputElement],
     HTMLLabelElement: [Function: HTMLLabelElement],
     HTMLLegendElement: [Function: HTMLLegendElement],
     HTMLLIElement: [Function: HTMLLIElement],
     HTMLLinkElement: [Function: HTMLLinkElement],
     HTMLMapElement: [Function: HTMLMapElement],
     HTMLMediaElement:
      { [Function: HTMLMediaElement]
        NETWORK_EMPTY: 0,
        NETWORK_IDLE: 1,
        NETWORK_LOADING: 2,
        NETWORK_NO_SOURCE: 3,
        HAVE_NOTHING: 0,
        HAVE_METADATA: 1,
        HAVE_CURRENT_DATA: 2,
        HAVE_FUTURE_DATA: 3,
        HAVE_ENOUGH_DATA: 4 },
     HTMLMenuElement: [Function: HTMLMenuElement],
     HTMLMetaElement: [Function: HTMLMetaElement],
     HTMLMeterElement: [Function: HTMLMeterElement],
     HTMLModElement: [Function: HTMLModElement],
     HTMLObjectElement: [Function: HTMLObjectElement],
     HTMLOListElement: [Function: HTMLOListElement],
     HTMLOptGroupElement: [Function: HTMLOptGroupElement],
     HTMLOptionElement: [Function: HTMLOptionElement],
     HTMLOutputElement: [Function: HTMLOutputElement],
     HTMLParagraphElement: [Function: HTMLParagraphElement],
     HTMLParamElement: [Function: HTMLParamElement],
     HTMLPreElement: [Function: HTMLPreElement],
     HTMLProgressElement: [Function: HTMLProgressElement],
     HTMLQuoteElement: [Function: HTMLQuoteElement],
     HTMLScriptElement: [Function: HTMLScriptElement],
     HTMLSelectElement: [Function: HTMLSelectElement],
     HTMLSourceElement: [Function: HTMLSourceElement],
     HTMLSpanElement: [Function: HTMLSpanElement],
     HTMLStyleElement: [Function: HTMLStyleElement],
     HTMLTableCaptionElement: [Function: HTMLTableCaptionElement],
     HTMLTableCellElement: [Function: HTMLTableCellElement],
     HTMLTableColElement: [Function: HTMLTableColElement],
     HTMLTableDataCellElement: [Function: HTMLTableDataCellElement],
     HTMLTableElement: [Function: HTMLTableElement],
     HTMLTableHeaderCellElement: [Function: HTMLTableHeaderCellElement],
     HTMLTimeElement: [Function: HTMLTimeElement],
     HTMLTitleElement: [Function: HTMLTitleElement],
     HTMLTableRowElement: [Function: HTMLTableRowElement],
     HTMLTableSectionElement: [Function: HTMLTableSectionElement],
     HTMLTemplateElement: [Function: HTMLTemplateElement],
     HTMLTextAreaElement: [Function: HTMLTextAreaElement],
     HTMLTrackElement: { [Function: HTMLTrackElement] NONE: 0, LOADING: 1, LOADED: 2, ERROR: 3 },
     HTMLUListElement: [Function: HTMLUListElement],
     HTMLUnknownElement: [Function: HTMLUnknownElement],
     HTMLVideoElement: [Function: HTMLVideoElement],
     StyleSheet: [Function: StyleSheet],
     MediaList: [Function: MediaList],
     CSSStyleSheet: [Function: CSSStyleSheet],
     CSSRule:
      { [Function: CSSRule]
        UNKNOWN_RULE: 0,
        STYLE_RULE: 1,
        CHARSET_RULE: 2,
        IMPORT_RULE: 3,
        MEDIA_RULE: 4,
        FONT_FACE_RULE: 5,
        PAGE_RULE: 6,
        KEYFRAMES_RULE: 7,
        KEYFRAME_RULE: 8,
        MARGIN_RULE: 9,
        NAMESPACE_RULE: 10,
        COUNTER_STYLE_RULE: 11,
        SUPPORTS_RULE: 12,
        DOCUMENT_RULE: 13,
        FONT_FEATURE_VALUES_RULE: 14,
        VIEWPORT_RULE: 15,
        REGION_STYLE_RULE: 16 },
     CSSStyleRule: { [Function: CSSStyleRule] parse: [Function] },
     CSSMediaRule: [Function: CSSMediaRule],
     CSSImportRule: [Function: CSSImportRule],
     CSSStyleDeclaration: [Function: CSSStyleDeclaration],
     StyleSheetList: [Function: StyleSheetList],
     XPathException: { [Function: XPathException] INVALID_EXPRESSION_ERR: 51, TYPE_ERR: 52 },
     XPathExpression: [Function: XPathExpression],
     XPathResult:
      { [Function: XPathResult]
        ANY_TYPE: 0,
        NUMBER_TYPE: 1,
        STRING_TYPE: 2,
        BOOLEAN_TYPE: 3,
        UNORDERED_NODE_ITERATOR_TYPE: 4,
        ORDERED_NODE_ITERATOR_TYPE: 5,
        UNORDERED_NODE_SNAPSHOT_TYPE: 6,
        ORDERED_NODE_SNAPSHOT_TYPE: 7,
        ANY_UNORDERED_NODE_TYPE: 8,
        FIRST_ORDERED_NODE_TYPE: 9 },
     XPathEvaluator: [Function: XPathEvaluator],
     HTMLCollection: [Function: HTMLCollection],
     NodeFilter:
      { [Function]
        acceptNode: [Function],
        SHOW_ALL: 4294967295,
        SHOW_ELEMENT: 1,
        SHOW_ATTRIBUTE: 2,
        SHOW_TEXT: 4,
        SHOW_CDATA_SECTION: 8,
        SHOW_ENTITY_REFERENCE: 16,
        SHOW_ENTITY: 32,
        SHOW_PROCESSING_INSTRUCTION: 64,
        SHOW_COMMENT: 128,
        SHOW_DOCUMENT: 256,
        SHOW_DOCUMENT_TYPE: 512,
        SHOW_DOCUMENT_FRAGMENT: 1024,
        SHOW_NOTATION: 2048,
        FILTER_ACCEPT: 1,
        FILTER_REJECT: 2,
        FILTER_SKIP: 3 },
     NodeIterator: [Function: NodeIterator],
     NodeList: [Function: NodeList],
     Blob: [Function: Blob],
     File: [Function: File],
     FileList: [Function: FileList],
     FormData: [Function: FormData],
     XMLHttpRequestEventTarget: [Function: XMLHttpRequestEventTarget],
     XMLHttpRequestUpload: [Function: XMLHttpRequestUpload],
     DOMTokenList: [Function: DOMTokenList],
     URL:
      { [Function: URL]
        domainToASCII: [Function: domainToASCII],
        domainToUnicode: [Function: domainToUnicode] },
     Window: [Function: Window] },
  _globalProxy: [Circular],
  __timers: { '0': [ [Object], [Function] ] },
  _top: [Circular],
  _parent: [Circular],
  _document:
   Document {
     location: [Getter/Setter],
     _reactListenersID36949607566930354: 0 },
  _sessionHistory: [ { document: [Object], url: [Object], stateObject: null } ],
  _currentSessionHistoryEntryIndex: 0,
  _length: 0,
  _virtualConsole:
   VirtualConsole {
     domain: null,
     _events: { error: [Function] },
     _eventsCount: 1,
     _maxListeners: undefined },
  length: 0,
  history: History {},
  parent: [Circular],
  frames: [Circular],
  document:
   Document {
     location: [Getter/Setter],
     _reactListenersID36949607566930354: 0 },
  self: [Circular],
  window: [Circular],
  top: [Circular],
  location: Location { valueOf: [Function: valueOf] },
  addEventListener: [Function: bound addEventListener],
  removeEventListener: [Function: bound removeEventListener],
  dispatchEvent: [Function: bound dispatchEvent],
  setTimeout: [Function],
  setInterval: [Function],
  clearInterval: [Function: bound stopTimer],
  clearTimeout: [Function: bound stopTimer],
  __stopAllTimers: [Function: bound stopAllTimers],
  Image: [Function],
  postMessage: [Function],
  atob: [Function],
  btoa: [Function],
  FileReader: { [Function: FileReader] EMPTY: 0, LOADING: 1, DONE: 2 },
  XMLHttpRequest: { [Function: XMLHttpRequest] UNSENT: 0, OPENED: 1, HEADERS_RECEIVED: 2, LOADING: 3, DONE: 4 },
  ArrayBuffer: [Function: ArrayBuffer],
  Int8Array: [Function: Int8Array],
  Uint8Array: [Function: Uint8Array],
  Uint8ClampedArray: [Function: Uint8ClampedArray],
  Int16Array: [Function: Int16Array],
  Uint16Array: [Function: Uint16Array],
  Int32Array: [Function: Int32Array],
  Uint32Array: [Function: Uint32Array],
  Float32Array: [Function: Float32Array],
  Float64Array: [Function: Float64Array],
  stop: [Function],
  close: [Function],
  getComputedStyle: [Function],
  console:
   { assert: [Function],
     clear: [Function],
     count: [Function],
     debug: [Function],
     error: [Function],
     group: [Function],
     groupCollapse: [Function],
     groupEnd: [Function],
     info: [Function],
     log: [Function],
     table: [Function],
     time: [Function],
     timeEnd: [Function],
     trace: [Function],
     warn: [Function] },
  navigator:
   { userAgent: [Getter],
     appName: [Getter],
     platform: [Getter],
     appVersion: [Getter],
     noUI: true,
     cookieEnabled: [Getter] },
  name: 'nodejs',
  innerWidth: 1024,
  innerHeight: 768,
  outerWidth: 1024,
  outerHeight: 768,
  pageXOffset: 0,
  pageYOffset: 0,
  screenX: 0,
  screenY: 0,
  screenLeft: 0,
  screenTop: 0,
  scrollX: 0,
  scrollY: 0,
  scrollTop: 0,
  scrollLeft: 0,
  screen: { width: 0, height: 0 },
  alert: [Function],
  blur: [Function],
  confirm: [Function],
  createPopup: [Function],
  focus: [Function],
  moveBy: [Function],
  moveTo: [Function],
  open: [Function],
  print: [Function],
  prompt: [Function],
  resizeBy: [Function],
  resizeTo: [Function],
  scroll: [Function],
  scrollBy: [Function],
  scrollTo: [Function],
  toString: [Function] }


```

**Comment by rafaelverger** [comment]
Any thoughts on this? I'm facing the same situation here :/ even with `window` and `document` set it throw this error :/

The render method is executed 3 times and then the error is thrown (probably because the first and the second times results in the same html)

**Comment by ljharb** [comment]
I'm going to close this, since it predates v3, and nobody's reported it in awhile. Please file a new issue if there's still problems.
